strawberryfields/gbs/similarity.py

```diff
@@ -201,3 +203,42 @@ def event_to_sample(photon_number: int, max_count_per_mode: int, modes: int) ->
             available_modes.remove(j)
 
     return sample
+
+
+def feature_vector_sampling(
+    samples: list, event_photon_numbers: list, max_count_per_mode: int = 2
+) -> np.ndarray:
+    """Calculates feature vector with respect to input samples.
+
+    The feature vector is composed of event probabilities, with all events having a maximum
+    photon count in each mode of ``max_count_per_mode``. Events are then described by their total
+    photon number, and those chosen as part of the feature vector can be specified through
+    ``event_photon_numbers``. Probabilities are reconstructed by measuring the occurrence of
+    events in the input ``samples``.
+
+    **Example usage**:
+
+    >>> feature_vector_sampling(samples, [2, 4, 6], 1)

```

Comment:
Yes :(
Was thinking we can do this really nicely if we get some pre-made samples.
But for now not sure what's best, felt like leaving it as made up was ok since `samples` is not visible.

Reply:
is this still WIP? In case it's not, ``samples`` should be defined in the example usage. Ideally, the example usage should be able to be copy and pasted by a user, and assuming they have their imports correct, everything should work.

Reply:
Good points, I've now improved things and the code is now executable (it's a bit slower though, maybe 30 seconds to run)